LSPS3/README.md

```diff
@@ -0,0 +1,317 @@
+  * `fee_base_msat` and `fee_proportional_millionths` are 0.
+  * `cltv_expiry_delta` is `lsp_cltv_expiry_delta`.
+
+All other fields are to be filled in by the client.
+
+The client MAY indicate multipart (`basic_mpp`) support, if it
+supports multipart.
+
+### 4.  Payment
+
+The client MUST connect to the LSP via an indicated
+`lsp_connection_strings`, and then issue the generated invoice
+to the payer of the client.
+
+The payer can then generate one or more outgoing payment parts to
+deliver the total amount to the LSP.
+
+The LSP, when it receives a forward where the next hop has an
+SCID matching an issued `jit_channel_scid`:
+
+* MUST wait for payment parts until all parts sum up to at least
+  `payment_size` plus `opening_fee`.
+* MUST NOT fail with `mpp_timeout`, as the final hop is the client
+  and that error is only allowed at the final hop.
+* MUST fail with `unknown_next_peer` if all parts have not
+  summed up to at least `payment_size` plus `opening_fee` AND
+  `opening_fee_valid_until` has passed.
+* MAY fail with `unknown_next_peer` if it receives too many payment
+  parts.
+* When all conditions below are true, MUST continue to the next
+  step:
+  * The current time is strictly less than `opening_fee_valid_until`.
+  * All received payment parts sum up to at least `payment_size`
+    plus `opening_fee`.
+  * The client is connected.
+
+### 5.  Channel Opening And Forwarding
+
+The LSP requests a channel open to the client.
+The LSP MUST set `option_zeroconf` in the channel opening.
+
+The channel size must be sufficient, including up to 1% LSP-side
+channel reserve, to forward the `payment_size` to the client.
+
+The LSP then forwards parts to the client as soon as both client
+and LSP have signalled `channel_ready` for the new channel.
+The LSP rewrites onions to deduct, in total, `opening_fee` from
+all parts it received, and to forward to the newly-opened channel.
+
+The LSP MUST ensure that each forwarded part forwards at least
+one millisatoshi.
+The LSP MUST ensure that all parts forwarded sum up to at least
+`payment_size`.
+
+For example, the LSP can use the algorithm below to determine
+how much to forward for each incoming payment part, while ensuring
+that both the above properties are preserved.
+
+* Put the payment parts in an array in arbitrary order.
+* For each part:
+  * Get the `part_amount`, the incoming amount for the part.
+  * If `part_amount - 1 >= opening_fee`, then forward
+    `part_amount - opening_fee` for this part, then forward
+    the rest of the parts verbatim, if any.
+  * Else subtract `opening_fee = opening_fee - (part_amount - 1)`,
+    then forward 1 millisatoshi for this part.
+* The above loop will not reach past the last part unless
+  the number of parts exceeds the minimum `payment_size`.
+
+Once the client has received payment parts that sum up to at
+least `payment_size`, it MUST claim at least one payment part.
+Once that is done, the protocol flow is complete.
```

Comment:
Explicitly add a section "Post-open Normal Opration" after this.  Explicitly mention that the 0-conf channel is opened with an SCID `alias` (`option_scid_alias`) and that subsequent invoices should use the `alias` and not the `jit_channel_scid`.